src/commonMain/kotlin/com/algolia/search/configuration/HttpClient.kt

```diff
@@ -44,11 +47,25 @@ internal fun HttpClientConfig<*>.configure(configuration: Configuration) {
         logger = Logger.SIMPLE
     }
     install(UserAgent) {
-        agent = "Algolia for Kotlin (${BuildConfig.version})"
+        agent =  "Algolia for Kotlin (${BuildConfig.version})"
     }
-    configuration.defaultHeaders?.let {
-        install(DefaultRequest) {
+    defaultRequest {
+        configuration.defaultHeaders?.let {
             it.forEach { (key, value) -> header(key, value) }
         }
+        if (method.canCompress()) {
+            compressionHeader(configuration.compression)
+        }
+    }
+}
+
+internal fun HttpMethod.canCompress(): Boolean {
+    return this == HttpMethod.Post || this == HttpMethod.Put
+}
+
+internal fun HttpRequestBuilder.compressionHeader(compression: Compression) {
+    when (compression) {
+        Compression.Gzip -> header(HttpHeaders.ContentEncoding, "gzip")
```

Comment:
Question: is the HTTP client setting the `Content-Length` header or do you have to set explicitly? (it's the case in Java)

Reply:
No it doesn't do it automatically, should I pass this header all the time, or only when doing gzip ?

Reply:
Answer is: always.

Reply:
My bad: Actually the http engine handles it.

Reply:
All good!